src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebefore/InputNoWhitespaceBeforeMethodRef.java

```diff
@@ -0,0 +1,23 @@
+package com . puppycrawl
+    .tools.
+    checkstyle.checks.whitespace.nowhitespacebefore;
+
+import java.util.function.Supplier;
+
+public class InputNoWhitespaceBeforeMethodRef {
+    public static class SomeClass {
+    public static class Nested<V> {
+      private Nested() {
+      }
+    }
+  }
+
+  public static class Nested2<V> {
+  }
+
+  public <V> void methodName(V value) {
+    Supplier<?> t = Nested2<V> ::new; //19:31 violation
+    Supplier<SomeClass.Nested<V>> passes = SomeClass.Nested ::new; //20:60 violation
```

Comment:
Just put violation, not the line/column.
If input is updated, no one will remember to change these positionings.